### PR TITLE
allow user to specify which directory they want to put their webjobs

### PR DESCRIPTION
**Pull request by watashiSHUN**
use "WEBJOB_PATH" in app_settings, related issue #2249

### CODE REVIEW

File: Kudu.Core/Environment.cs [hunk 1]
```diff
@@ -117,6 +117,13 @@ public class Environment : IEnvironment
             _dataPath = Path.Combine(rootPath, Constants.DataPath);
             _jobsDataPath = Path.Combine(_dataPath, Constants.JobsPath);
             _jobsBinariesPath = Path.Combine(_webRootPath, Constants.AppDataPath, Constants.JobsPath);
+            // if userDefinedWebJobRoot is relative path, append it to site/wwwroot/ else use it as an absolute path
```
Comment (ahmelsayed) [inline Kudu.Core/Environment.cs]: the code doesn't seem to do what this comment says

File: Kudu.Core/Environment.cs [hunk 2]
```diff
@@ -1,13 +1,13 @@
-﻿using System;
+﻿using Kudu.Contracts.Settings;
```
Comment (ahmelsayed) [inline Kudu.Core/Environment.cs]: I believe we order but put all 'using System's at the top. It's one of the options in VS

File: Kudu.Core/Environment.cs (hunk 1 above)
Comment (watashiSHUN) [inline Kudu.Core/Environment.cs]: `_webRootPath = "D:/home/site/wwwroot/"`

1. default `_jobsBinariesPath = "D:/home/site/wwwroot/App_Data/jobs"`
2. if `userDefinedWebJobRoot = "myfunctions"`, `_jobsBinariesPath = "D:/home/site/wwwroot/myfunctions"`
3. if `userDefinedWebJobRoot = "D:/home/functionfolder"`, `_jobsBinariesPath = "D:/home/functionfolder"`

Path.Combine(path1,path2) returns path2 if path2 is an absolute path

Comment (ahmelsayed) [inline Kudu.Core/Environment.cs]: I see. Maybe add that to the comment and move it above L125.

**Comment by ahmelsayed** [comment]
looks good. It'd be nice if you could add a unit test for it, but it can be in another commit :shipit:

**Comment by davidebbo** [comment]
Related thread: https://social.msdn.microsoft.com/Forums/azure/en-US/11549943-9aac-4e0f-aa7d-207d3d40cfcc/iif-you-change-the-directory-mapping-of-root-web-site-in-web-apps-webjobs-deployment-gets?forum=windowsazurewebsitespreview

**Comment by davidebbo** [comment]
@watashiSHUN it sounds like one was reviewed and is good to go. If so, can you squash/merge it in?

**Comment by watashiSHUN** [comment]
Done, reminder to self, update kudu documentation

**Comment by davidebbo** [comment]
@watashiSHUN it doesn't look like it was merged, right?

For the doc, add it under https://github.com/projectkudu/kudu/wiki/WebJobs#configuration-settings

**Comment by watashiSHUN** [comment]
https://github.com/projectkudu/kudu/commit/581903cd573b4c37045ff42323c441773f66ed66 looks like my commit was sandwiched by @snobu, so didn't see it?

**Comment by davidebbo** [comment]
Ah yes, my bad! Ok, just add the wiki doc and close then.

**Comment by watashiSHUN** [comment]
maybe rename it to **WEBJOB_TARGET_PATH** / **WEBJOB_SOURCE_PATH**? there's a very similar **WEBJOBS_PATH** environment variable in the [doc](https://github.com/projectkudu/kudu/wiki/WebJobs#environment-settings)

**Comment by davidebbo** [comment]
Let's go with `WEBJOBS_ROOT_PATH`, as that . Note that it should be WEBJOB**S** and now WEBJOB.

**Pull request closed without merging**